code/controlconfig/controlsconfigcommon.cpp

```diff
@@ -771,7 +773,7 @@ const char *textify_scancode(int code)
 		}
 	}
 
-	strcat_s(text, Scan_code_text[keycode]);
+	strcat_s(text, SDL_GetKeyName(SDL_GetKeyFromScancode(FS2toSDL[keycode])));
```

Comment:
std::map::operator[] will return an empty value if the given key is not already in the map, in addition to adding the new key and empty value pair to the map.  You must either ensure `keycode` is already mapped or use ::at()

Reply:
considering that we need to actually handle the case that keycode not being in FS2toSDL should return SDL_SCANCODE_UNKNOWN, I've replaced this with a find and a check if this returns map::end instead of ::at which would throw an exception.
But the insertion behaviour of [] would certainly be undesirable